File: src/test/java/komu/blunt/types/TypeApplicationTest.java
package komu.blunt.types;

import org.junit.Test;

import static komu.blunt.types.Type.functionType;
import static komu.blunt.types.Type.typeVariable;
import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class TypeApplicationTest {

    @Test
    public void printingArrowTypes() {
        assertThat(functionType(typeVariable("a"), typeVariable("b")).toString(), is("a -> b"));
    }
}

File: src/test/kotlin/komu/blunt/types/TypeApplicationTest.kt
package komu.blunt.types

import org.junit.Test
import kotlin.test.assertEquals

class TypeApplicationTest {

    @Test
    fun printingArrowTypes() {
        //TypeVariable("a", Kind.STAR)
        assertEquals("foo", "foo")
        //functionType(typeVariable("a"), typeVariable("b"))
        //assertEquals("a -> b", functionType(typeVariable("a"), typeVariable("b")).toString())
    }
}
